index.js

```diff
@@ -227,4 +227,39 @@ transform.resetTypes = function resetTypes() {
   registerTypes = {};
 };
 
-module.exports = transform;
+function getFormOptions(permission, options, fields, properties, template) {
+  const newOptions = []
+  Object.entries(properties).forEach((data) => {
+      const [key, value] = data
+      newOptions[key] = {
+          fields: [],
+          order: fields[key].order
+      }
+      Object.keys(value.properties).forEach((subKey) => {
```

Comment:
This assume that there are only two levels of nesting and that all the fields are on the second level. That will not work in many scenarios.
Examples:
This schema would be fine:
```
{
  "properties": {
    "Hours": {
      "type": "object",
      "properties": {
        "st": {
          "type": "number"
        },
        "ot": {
          "type": "number"
        }
      }
    }
  }
}
```
This  schema will not work with this code:
```
{
  "properties": {
    "Hours": {
      "type": "object",
      "properties": {
        "st": {
          "type": "number"
        },
        "overtime hours": {
          "type": "object",
          "properties": {
            "ot": {
              "type": "number"
            },
            "pot": {
              "type": "number"
            }
          }
        }
      }
    }
  }
}
```
This schema will also not work with this code:
```
{
  "properties": {
    "st": {
      "type": "number"
    }
  }
}
```

Instead this should allow any level of nesting of objects. There are multiple ways to do that, but I'd suggest pulling out the logic to a recursive function which gets the the options for one property (and all its sub properties by calling itself recursively).

Reply:
For an example of a recursive function solving a similar problem, see `mapUISchemaComponentToColDefs`.

Reply:
yeah, I see. I need to figured out the best way to do this.